frontends/pytorch/python/torch_mlir_torchscript/e2e_test/reporting.py

```diff
@@ -54,34 +54,114 @@ def format_error(self, s: str):
+            if value != golden:
+                return self._record_failure(
+                    f'value ({value!r}) is not equal to golden value ({golden!r})'
+                )
+            return
+        if isinstance(golden, str):
+            if not isinstance(value, str):
+                return self._record_mismatch_type_failure('str', value)
+            if value != golden:
+                return self._record_failure(
+                    f'value ({value!r}) is not equal to golden value ({golden!r})'
+                )
+            return
+        if isinstance(golden, tuple):
+            if not isinstance(value, tuple):
+                return self._record_mismatch_type_failure('tuple', value)
+            if len(value) != len(golden):
+                return self._record_failure(
+                    f'value ({len(value)!r}) is not equal to golden value ({len(golden)!r})'
+                )
+            reports = [
+                ValueReport(v, g, self.context.chain(f'tuple element {i}'))
+                for i, (v, g) in enumerate(zip(value, golden))
+            ]
+            for report in reports:
+                if report.failed:
+                    self.failure_reasons.extend(report.failure_reasons)
+            return
+        if isinstance(golden, list):
+            if not isinstance(value, list):
+                return self._record_mismatch_type_failure('list', value)
+            if len(value) != len(golden):
+                return self._record_failure(
+                    f'value ({len(value)!r}) is not equal to golden value ({len(golden)!r})'
+                )
+            reports = [
+                ValueReport(v, g, self.context.chain(f'list element {i}'))
+                for i, (v, g) in enumerate(zip(value, golden))
+            ]
+            for report in reports:
+                if report.failed:
+                    self.failure_reasons.extend(report.failure_reasons)
+            return
+        if isinstance(golden, dict):
+            if not isinstance(value, dict):
+                return self._record_mismatch_type_failure('dict', value)
+            gkeys = list(sorted(golden.keys()))
+            vkeys = list(sorted(value.keys()))
+            if gkeys != vkeys:
+                return self._record_failure(
+                    f'dict keys ({vkeys!r}) are not equal to golden keys ({gkeys!r})'
+                )
+            reports = [
+                ValueReport(value[k], golden[k],
+                            self.context.chain(f'dict element at key {k!r}'))
+                for k in gkeys
+            ]
+            for report in reports:
+                if report.failed:
+                    self.failure_reasons.extend(report.failure_reasons)
+            return
+        if isinstance(golden, torch.Tensor):
+            if not isinstance(value, torch.Tensor):
+                return self._record_mismatch_type_failure('torch.Tensor', value)
```

Comment:
allclose would throw an exception if the tensor sizes mismatch so

Reply:
Good catch!!